extensions/amp-analytics/0.1/events.js

```diff
@@ -306,6 +332,83 @@ export class EventTracker {
   add(unusedContext, unusedEventType, unusedConfig, unusedListener) {}
 }
 
+/**
+ * Tracks custom browser events.
+ */
+export class CustomBrowserEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.observables_ = new Observable();
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.observables_.fire.bind(this.observables_);
+  }
+
+  /** @override */
+  dispose() {
+    const root = this.root.getRoot();
+    Object.keys(BrowserEventType).forEach((key) => {
+      root.removeEventListener(BrowserEventType[key], this.boundOnSession_);
+    });
+    this.boundOnSession_ = null;
+    this.observables_ = null;
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    const selector = userAssert(
+      config['selector'],
+      'Missing required selector on browser event trigger'
+    );
+    userAssert(
+      selector.length,
+      'Missing required selector on browser event trigger'
+    );
+    assertUniqueSelectors(selector);
+
+    const selectionMethod = config['selectionMethod'] || null;
+    const eventName = config['on'];
+
+    const targetReady = this.root.getElements(
+      context,
+      selector,
+      selectionMethod,
+      false
+    );
+
+    debounce(
```

Comment:
This is not the right way to use `debounce`. Couple of things you have wrong here:
1. You want to debounce the callback being passed in the `addEventListener`
2. `debounce` returns a function, so you typically want to use assign it to a var & pass into as an argument to `addEventListener`